### ISSUE TITLE
MAC vendor (OUI) database out of date? 

### ISSUE THREAD

**Issue by RedBearAK** [angryip/ipscan#317]
On Ubuntu 21.10, ran into the usual "no arp installed" issue, so no MAC addresses or vendors initially, but fixed that with "apt install net-tools". Now all devices that appear after a scan at least show the MAC address. But there is some inconsistency of the MAC address being translated into a MAC vendor. 

I have three different Synology NAS devices, but only two show a MAC vendor: 

```
00:11:32:xx:xx:xx - Synology
00:11:32:xx:xx:xx - Synology
90:09:D0:xx:xx:xx - [n/a]
```

I only knew the last one was Synology because of the hostname and NetBIOS info, which reflected the descriptive name I gave it (i.e., "synods220plus.local"). 

This 90:09:D0 address prefix shows up in some online MAC vendor lookup sites correctly as "Synology", but doesn't appear in at least one I checked. Might be a relatively recent addition. The device is a 2020 model, DS220+. 

I looked up "ieee mac vendor database" and the first link goes here: 

http://standards-oui.ieee.org/oui/oui.txt

This file shows both Synology MAC address prefixes detected on my network as having the same "Synology Incorporated" information attached: 

```
00-11-32   (hex)		Synology Incorporated
001132     (base 16)		Synology Incorporated
				9F., No.1, Yuandong Rd., Banqiao Dist.,
				New Taipei City    220632
				TW

90-09-D0   (hex)		Synology Incorporated
9009D0     (base 16)		Synology Incorporated
				9F., No.1, Yuandong Rd., Banqiao Dist.,
				New Taipei City    220632
				TW
```

I also have an EC-2B-EB device that is reporting `[n/a]` instead of Amazon. This, too,  is listed in the current IEEE `oui.txt` file at the link above. 

The only other device detected on my network that is not reporting a MAC vendor is something with a prefix of 0A-46-E5, which I can find no reference to anywhere, including the IEEE OUI file. So that is at least understandable. But it should be the only device that doesn't show a MAC vendor. 

And there is something appearing with a MAC vendor of "TENOTECHNOLOGIES", which is a string that doesn't exist in the IEEE OUI file. But there is this: 

```
E0-B2-60   (hex)		TENO NETWORK TECHNOLOGIES COMPANY LIMITED
E0B260     (base 16)		TENO NETWORK TECHNOLOGIES COMPANY LIMITED
				RM 1302, 13/F CHEONG K BLDG 84-86 DES VOEUX RD CENTRAL  HONG KONG
				Hong Kong    999077
				HK
```

If `ipscan` is just pulling MAC vendor information from a local file, with no provision for searching online or automatically updating the vendor database, maybe it's at least time to update the copy of the OUI database that you include with new versions? If this is in fact the source of the vendor list you include with the distributed packages. Just guessing based on comments I've seen on some other issues here. 

Couldn't find the `oui.txt` file installed on my system so I looked inside the JAR and found `mac-vendors.txt`. It appears to be a simplified list of MAC addresses and vendors. Last updated Feb. 2021. 

There are 29,498 lines in the existing mac-vendors.txt file. Creating a similar filtered and sorted text file from the current IEEE OUI text file I wound up with a text file with 31,209 lines. Seems there may be 1,711 MAC address prefixes and vendors missing from the file included with the most current release of `ipscan`. 

There is also a text encoding error reported by the Nautilus GNOME Sushi preview plugin when I attempt to preview the mac-vendors.txt file, or when I try to open it with the GNOME text editor. I find this strange, as this is the first file I've ever opened with the GNOME text editor that warned me about the text encoding, and that I might "damage the file" if I continued editing it. Shouldn't this file be using a UTF-8 format by now? I had to change to the older ISO-8859-1 text encoding to get the text editor to stop complaining. 

So the text encoding format of this new generated file, besides not having quite the same simplified descriptions, may not be compatible with your app in its current state. I'm just presenting it here as an example: 

[mac-vendors-new.txt](https://github.com/angryip/ipscan/files/7782228/mac-vendors-new.txt)

**Comment by angryziber** [comment]
For future reference: there is [update-mac-vendors.sh](https://github.com/angryip/ipscan/blob/master/update-mac-vendors.sh) script that is used to update mac vendors before each release.

**Comment by RedBearAK** [comment]
@angryziber 

Fantastic, thanks. 
Happy New Year! 
![image](https://user-images.githubusercontent.com/64876997/147840433-7b674367-5417-48c8-8418-d8f24389cd67.png)

**Comment by RedBearAK** [comment]
@angryziber 

FYI, I figured out why some devices are still not showing up with a known MAC vendor, even with the updated file. They were Apple mobile devices with the "Private Address" setting enabled, which randomizes the MAC address for each network the device connects to. 

When I turn the setting off, the device(s) will then show up as "Apple", of course. 

Info about Apple's "Private Address" setting for reference: 

https://support.apple.com/en-us/HT211227

I found out by using `nmap` to reveal that the only open port at those IP addresses was `62078/tcp` (labeled by nmap as "iphone sync?").
